application/F3DStarter.cxx

```diff
@@ -47,6 +47,33 @@ class F3DStarter::F3DInternals
     {
       cam.setViewAngle(appOptions.CameraViewAngle);
     }
+
+    bool reset = false;
+    double zoomFactor = 0.9;
+    if (appOptions.CameraPosition.size() != 3 && appOptions.CameraDirection.size() == 3)
+    {
+      f3d::vector3_t dir;
+      std::copy_n(appOptions.CameraDirection.begin(), 3, dir.begin());
+      f3d::point3_t foc;
+      f3d::point3_t pos;
+      cam.getFocalPoint(foc);
+      for (int i = 0; i < 3; i++)
+      {
+        pos[i] = foc[i] - dir[i];
+      }
+      cam.setPosition(pos);
+      reset = true;
+    }
+    if (appOptions.CameraPosition.size() != 3 && appOptions.CameraZoomFactor != 0)
```

Comment:
what happens if `CameraZoomFactor` is negative?

Reply:
it is ignored by VTK, but lets ignore it at F3D level instead.